src/Microsoft.Framework.PackageManager/List/DependencyListCommand.cs

```diff
@@ -1,40 +1,36 @@
 ﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
-using System.Linq;
 using System.Runtime.Versioning;
+using Microsoft.Framework.PackageManager.Utils;
 
 namespace Microsoft.Framework.PackageManager.List
 {
     internal class DependencyListCommand
     {
         private readonly DependencyListOptions _options;
+        private readonly FrameworkName _fallbackFramework;
 
-        public DependencyListCommand(DependencyListOptions options)
+        public DependencyListCommand(DependencyListOptions options, FrameworkName fallbackFramewok)
         {
             _options = options;
+            _fallbackFramework = fallbackFramewok;
         }
 
         public int Execute()
         {
-            var result = 0;
             _options.Reports.Information.WriteLine("Listing dependencies for {0} ({1})", _options.Project.Name, _options.Project.ProjectFilePath);
 
-            var frameworks = new HashSet<FrameworkName>(_options.Project.GetTargetFrameworks().Select(f => f.FrameworkName));
-            if (_options.Framework != null)
+            string frameworkSelectionErrors;
+            var frameworks = FrameworkSelectionHelper.SelectFrameworks(_options.Project, 
+                                                                       _options.TargetFrameworks, 
+                                                                       _fallbackFramework, 
+                                                                       out frameworkSelectionErrors);
```

Comment:
nit: `frameworkSelectionErrors` => `error` or `frameworkSelectionError`, If you always populate a single error.